Add stardist fluorescence model

gnodar01 opened the pull request:
We have had stardist H&E for a while, which is pretrained for tissue images, this adds the variant of stardist pretrained on flourescence data from here: https://bioimage.io/#/?tags=stardist&id=10.5281%2Fzenodo.6348084

gnodar01 committed:
[add] Add stardist fluorescence model

- we have had stardist H&E for a while, which is pretrained for tissue
  images, this adds the variant of stardist pretrained on flourescence
  data from here: https://bioimage.io/#/?tags=stardist&id=10.5281%2Fzenodo.6348084
- TODO: DRY the code, by unifying preprocessing, prediction and loading
  from the two stardist implementations (majority is identical between
  the two)

gnodar01 committed:
[mod] Update documentation and folder names

gnodar01 committed:
[opt] Make Stardist code DRY

- create abstract stardist class for the two concrete stardist variant
  classes, because most of the functionality is identical

gnodar01 committed:
[mod] Make minor updates to comments

Andrea-Papaleo commented:
Running into error while predicting.
`Error in running predictions
Error: The shape of dict['input'] provided in model.execute(dict) must be [-1,-1,-1,1], but was [1,128,128,3]`

![Stardist Example Fluo](https://github.com/piximi/piximi-beta/assets/93597832/462719cf-8d16-4a53-8cd2-6289d13c08b0)

gnodar01 committed:
[fix] Set req channels to 1

gnodar01 commented:
The fluorescence model takes one channel inputs. You can find the example image at the [zenodo repo](https://zenodo.org/records/6348085), but don't download the `jpg` (since that's a scaled down, 3 channel, preview image). You should download `sample_input_0.tif`, which is one channel and 32-bit. Since 32-bit images are [not yet supported](https://github.com/piximi/piximi-beta/blob/master/src/utils/common/tests/tensorHelper.test.ts#L348-L349) you have to scale it down to 16-bit in fiji via `Image -> type -> 16-bit` (I can't upload it here since GitHub doesn't support `tiff`).

UPDATE: I added the example image in `data/images`.

I did make the mistake of setting `3` for `requiredChannels` in the model constructor, which I've now fixed, but there is no difference in behavior if passing in the 3 channel image. The same error is issued; at least on predict piximi doesn't stop you from trying to press `predict` on a 3 channel image. Filed an issue [here](https://github.com/piximi/piximi-beta/issues/4).

gnodar01 committed:
[Add] Add example image for stardist fluo
